exec/java-exec/src/test/java/org/apache/drill/TestFunctionsQuery.java

```diff
@@ -923,4 +923,40 @@ public void testRandom() throws Exception {
             .baselineValues(false)
             .go();
   }
+
```

Comment:
Let's create new test class for `date_add` similar to `TestDateTruncFunctions` and test there all interval variations (year, month, hour etc).